Tests/NetworkServiceTests/HTTPTaskTests.swift

```diff
@@ -0,0 +1,127 @@
+import NetworkService
+import XCTest
+
+final class HTTPTaskTests: XCTestCase {
+    func testHTTPTaskCasesPresentInAPI() {
+        _ = HTTPTask.request
+        _ = HTTPTask.requestWithURLParameters(urlParameters: nil)
+        _ = HTTPTask.requestWithBodyParameters(bodyParameters: nil)
+    }
+
+    func testCreatesHTTPTaskWithEmptyURLParameters() {
+        // Given
+        let emptyUrlParameters = URLParameters()
+
+        // When
+        let task = HTTPTask.requestWithURLParameters(urlParameters: emptyUrlParameters)
+
+        // Then
+        XCTAssertNotNil(task)
+    }
```

Comment:
Maybe it makes sense to check that the expected type of object was created instead of only the task was created. Please review the rest of code.
```suggestion
        // Then
              XCTAssertNotNil(task as HTTPTask)
    }
```

Reply:
I'm not sure that it makes sense. We know type of the task in compile time. Why could¡¡ we expect another result?
<img width="1068" alt="Screenshot 2023-08-04 at 15 24 28" src="https://github.com/ios-course/simple-network-service/assets/110284165/440cb083-1f3a-4ce3-a7c8-5d28fb8141e4">
On the other hand we know that the `task` isn't optional `HTTPTaks` and it doesn't have the sense to check on the `nil`.
I did in another way, check it.